EventFiltering/PWGHF/HFFilter.cxx

```diff
@@ -249,8 +249,60 @@ DECLARE_SOA_TABLE(HFTrigTrain3P, "AOD", "HFTRIGTRAIN3P", //!
                   hftraining3p::FlagOrigin,
                   hftraining3p::Channel,
                   hftraining3p::HFSelBit);
+
+namespace hfoptimisationTree2Prong
+{
+DECLARE_SOA_COLUMN(CollisionIndex, collisionIndex, int);           //!
+DECLARE_SOA_COLUMN(BkgBDTD0, bkgBDTD0, float);           //!
+DECLARE_SOA_COLUMN(PromptBDTD0, promptBDTD0, float);           //!
+DECLARE_SOA_COLUMN(NonpromptBDTD0, nonpromptBDTD0, float);           //!
+DECLARE_SOA_COLUMN(DCAXYD0, dcaXYD0, float);           //!
+} // namespace hfoptimisationTree2Prong
+DECLARE_SOA_TABLE(HFOptimisationTree2Prong, "AOD", "HFOPTIMTREE2P", //!
+                  hfoptimisationTree2Prong::CollisionIndex,
+                  hfoptimisationTree2Prong::BkgBDTD0,
+                  hfoptimisationTree2Prong::PromptBDTD0,
+                  hfoptimisationTree2Prong::NonpromptBDTD0,
+                  hfoptimisationTree2Prong::DCAXYD0);
+
+namespace hfoptimisationTree3Prong
```

Comment:
If we don't have 2-prong or 3-prong specific cuts, we do not need to define 2 tables: it can be only one (and we can have in one column the "identity" of the candidate, using for example the `beautyParticles` enum)

Reply:
Yes this would be great!

Reply:
In the loops, we have access to the iterator index (`iHypo`) but I don't know how to get the string inside `beautyParticles` enum from this.
Actually, it depends on the kind of ID we want:

- an `int`: in this case I just need to drop the index corresponding to the index of the particle in `beautyParticles` (but it would require to know which index corresponds to which particle afterwards)
- a `std::string` but this would require (in the case of Bplus) to be able to get the `kBplus` from `beautyParticles` if the index is `0`.